module/os/freebsd/zfs/kmod_core.c

```diff
@@ -133,18 +133,15 @@ zfsdev_ioctl(struct cdev *dev, ulong_t zcmd, caddr_t arg, int flag,
 	len = IOCPARM_LEN(zcmd);
 	vecnum = zcmd & 0xff;
 	zp = (void *)arg;
-	uaddr = (void *)zp->zfs_cmd;
 	error = 0;
 #ifdef ZFS_LEGACY_SUPPORT
 	zcl = NULL;
 #endif
 
-	if (len != sizeof (zfs_iocparm_t)) {
-		printf("len %d vecnum: %d sizeof (zfs_cmd_t) %ju\n",
-		    len, vecnum, (uintmax_t)sizeof (zfs_cmd_t));
+	if (len != sizeof (zfs_iocparm_t))
 		return (EINVAL);
-	}
 
+	uaddr = (void *)zp->zfs_cmd;
```

Comment:
Why was this moved to a new line?

Reply:
Because before the len is checked, zp may theoretically point to a single byte.

Reply:
I hadn't even noticed that - good catch @amotin